app/src/utils/socket.js

```diff
@@ -0,0 +1,52 @@
+import io from 'socket.io-client'
+
+const host = 'wss://odyssey.wancat.cc'
+
+class SocketWrapper {
+  constructor() {
+    this.socket = null
+    this.listeners = []
+  }
+
+  connect(session) {
+    if (this.socket) {
+      this.socket.off()
+      this.socket.disconnect()
+    }
+
+    this.socket = io(host, {
+      query: {
+        session_id: session._id,
+      },
+    })
+
+    this.socket.on('connect', () => {
+      console.log('Socket connected to session', session._id)
+    })
+
+    this.reconnect()
+  }
```

Comment:
It's not "reconnect in socket.on" but reconnect after connect to server. Because we need to reset all the callbacks other components use. But "reconnect" may be confusing, so I replace it to "bindListeners" instead